ompi/request/req_test.c

```diff
@@ -338,30 +338,46 @@ int ompi_request_default_test_some(
     int * indices,
     ompi_status_public_t * statuses)
 {
-    size_t i, num_requests_null_inactive = 0, num_requests_done = 0;
+    size_t i, num_requests_null_inactive, num_requests_done;
     int rc = OMPI_SUCCESS;
     ompi_request_t **rptr;
     ompi_request_t *request;
-
     opal_atomic_mb();
+
+#if OPAL_ENABLE_PROGRESS_THREADS == 0
+    int do_it_once = 0;
+#endif
+
+    num_requests_null_inactive = 0;
+    num_requests_done = 0;
     rptr = requests;
     for (i = 0; i < count; i++, rptr++) {
         request = *rptr;
+#if OPAL_ENABLE_PROGRESS_THREADS == 0
+recheck_request_status:
```

Comment:
I think you need to reset num_requests_null_inactive and num_requests_done to 0 when taking the goto